### PR TITLE
Fix ctrl-c exit on Windows and add --version flag

### PR DESCRIPTION
**Pull request by 3tilley**
This is a really cool utility!

I've made an update to make ctrl-c work on Windows. There is technically a breaking change because now on Linux the return code is 130, while previously it was 2. If you don't attach the ctrl-c handler on Linux (at least in Bourne Shell derived shells) the exit code would have been 130, so I think it's "technically" more correctly, if that really matters.

I've also given it a `--version` command, which was useful for me while messing around with it locally. If you're happy with the change, would you like me to bump the version within this PR?

Additionally, are you open to adding more features to this tool? I used something homegrown previously that would print out the time of last commit, the author, and a flag to read either local or remote branches for changes. I could do knock up a PR to do something similar here, or fork for my own use if that's not what you'd intended for this crate.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -104,6 +119,9 @@ fn run_tui() -> Result<()> {
 fn dialoguer_reset_cursor_hack() {
     let term = dialoguer::console::Term::stdout();
     let _ = term.show_cursor();
+    // Exit 130 is the standard exit code for a process that has been interrupted (128 + SIGINT)
+    process::exit(130);
+
```
Comment (dnjstrom) [inline src/main.rs]: Nice catch! 🌟

File: src/main.rs [hunk 2]
```diff
@@ -17,13 +17,28 @@ use config::Config;
...
+        1 => match run_tui() {
+            Ok(()) => Ok(()),
+            Err(e) => match e.downcast_ref::<SelectBranchError>() {
+                Some(SelectBranchError::Aborted) => process::exit(1),
+                Some(SelectBranchError::Interrupted) => process::exit(2),
+                None => Err(e),
+            },
         },
+        2 => {
+           if ( args[1] == "--version" ) | ( args[1] == "-V" ) {
```
Comment (dnjstrom) [inline src/main.rs]: In my experience, a lower-case "v" is the more common short-hand (though to be fair, I almost exclusively use unix-systems). I wouldn't be opposed to making the flags fully case insensitive if you prefer that?

```suggestion
           if ( args[1] == "--version" ) | ( args[1] == "-v" ) {
```

File: src/main.rs [hunk 3]
```diff
@@ -17,13 +17,28 @@ use config::Config;
...
+               println!("{}", env!("CARGO_PKG_VERSION"));
+               Ok(())
+           } else {
+               Err(Error::msg(format!("Command not recognised {}", args[1])))
+           }
+        }
+        _ =>
+            {
+                Err(Error::msg(format!("Commands not recognised {:?}", &args[1..])))
```
Comment (dnjstrom) [inline src/main.rs]: I'd argue even if multiple arguments are passed, it's still one command 🤷 

```suggestion
                Err(Error::msg(format!("Command not recognized {:?}", &args[1..])))
```

File: src/main.rs [hunk 4]
```diff
@@ -17,13 +17,28 @@ use config::Config;
...
+                Some(SelectBranchError::Interrupted) => process::exit(2),
+                None => Err(e),
+            },
         },
+        2 => {
+           if ( args[1] == "--version" ) | ( args[1] == "-V" ) {
+               println!("{}", env!("CARGO_PKG_VERSION"));
+               Ok(())
+           } else {
+               Err(Error::msg(format!("Command not recognised {}", args[1])))
```
Comment (dnjstrom) [inline src/main.rs]: I tend to prefer American english spellings, so for consistency let's go with that here

```suggestion
               Err(Error::msg(format!("Command not recognized {}", args[1])))
```

**Review by dnjstrom — CHANGES_REQUESTED** [review]
Thank you for the PR! I'm happy that you're enjoying git-select-branch! 🌟 

You're absolutely right that a 130 status is more correct - good catch! Haven't really tried this on Windows but great to know it (almost) works! The `--version` flag addition is awesome too, thanks for setting it up 🙌 

As for bumping the version, if you would please go ahead and submit that in this PR that'd be swell!

As for additional features, I tend to opt for small and focused tools (following the unix philosophy) but if you have an idea that you think would fit well under the conceptual umbrella of "selecting a git branch" I'd be happy to discuss it! Please open an issue with some simple text mockups of how it might work (so you don't put a bunch of effort into an implementation that ultimately won't be accepted) and we can take it from there 👍

**Comment by 3tilley** [comment]
Hello, thanks for your thorough response and great review! I agree with all of the above, I'll make the changes.

Regarding `-v / -V` it's definitely personal, I generally use `-v / -vv / -vvv` for verbosity of output, which some other tools do. I've just run it on my cargo bin folder to show some usecases, it's a bit split. Should we keep to just `--version` for now? Note that anything that prints a help message indicates unrecognised arg.
<details> 
  <summary>--version / -v / -V output</summary>

```
bat.exe
----------
bat.exe --version
bat 0.22.1
bat.exe -v
error: Found argument '-v' which wasn't expected, or isn't valid in this context

	If you tried to supply `-v` as a value rather than a flag, use `-- -v`

USAGE:
    bat.exe [OPTIONS] [FILE]...
    bat.exe <SUBCOMMAND>

For more information try --help
bat.exe -V
bat 0.22.1
broot.exe
----------
broot.exe --version
broot 1.26.2-dev
broot.exe -v
error: unexpected argument '-v' found

  tip: to pass '-v' as a value, use '-- -v'

Usage: broot.exe [OPTIONS] [ROOT]
broot.exe -V
error: unexpected argument '-V' found

  tip: to pass '-V' as a value, use '-- -V'

Usage: broot.exe [OPTIONS] [ROOT]
cargo-clippy.exe
----------
cargo-clippy.exe --version
clippy 0.1.73 (cc66ad46 2023-10-03)
cargo-clippy.exe -v
error: could not find `Cargo.toml` in `C:\Users\Max\.cargo\bin` or any parent directory
cargo-clippy.exe -V
clippy 0.1.73 (cc66ad46 2023-10-03)
cargo-expand.exe
----------
cargo-expand.exe --version
cargo-expand 1.0.72
cargo-expand.exe -v
error: unexpected argument '-v' found

Usage: cargo <COMMAND>

For more information, try '--help'.
cargo-expand.exe -V
cargo-expand 1.0.72
cargo-fmt.exe
----------
cargo-fmt.exe --version
rustfmt 1.6.0-stable (cc66ad46 2023-10-03)
cargo-fmt.exe -v
`cargo metadata` exited with an error: error: could not find `Cargo.toml` in `C:\Users\Max\.cargo\bin` or any parent directory

This utility formats all bin and lib files of the current crate using rustfmt.

Usage: cargo fmt [OPTIONS] [-- <rustfmt_options>...]

Arguments:
  [rustfmt_options]...  Options passed to rustfmt

Options:
  -q, --quiet
          No output printed to stdout
  -v, --verbose
          Use verbose output
      --version
          Print rustfmt version and exit
  -p, --package <package>...
          Specify package to format
      --manifest-path <manifest-path>
          Specify path to Cargo.toml
      --message-format <message-format>
          Specify message-format: short|json|human
      --all
          Format all packages, and also their local path-based dependencies
      --check
          Run rustfmt in check mode
  -h, --help
          Print help
cargo-fmt.exe -V
error: unexpected argument '-V' found

  tip: to pass '-V' as a value, use '-- -V'

Usage: cargo fmt [OPTIONS] [-- <rustfmt_options>...]

For more information, try '--help'.
cargo-miri.exe
----------
cargo-miri.exe --version
error: the 'miri' component which provides the command 'cargo-miri.exe' is not available for the 'stable-x86_64-pc-windows-msvc' toolchain
cargo-miri.exe -v
error: the 'miri' component which provides the command 'cargo-miri.exe' is not available for the 'stable-x86_64-pc-windows-msvc' toolchain
cargo-miri.exe -V
error: the 'miri' component which provides the command 'cargo-miri.exe' is not available for the 'stable-x86_64-pc-windows-msvc' toolchain
cargo-shuttle.exe
----------
cargo-shuttle.exe --version
cargo-shuttle 0.29.1
cargo-shuttle.exe -v
error: unexpected argument '-v' found

  tip: to pass '-v' as a value, use '-- -v'

Usage: cargo-shuttle.exe [OPTIONS] <COMMAND>

For more information, try '--help'.
cargo-shuttle.exe -V
cargo-shuttle 0.29.1
cargo-tauri.exe
----------
cargo-tauri.exe --version
tauri-cli 1.2.3
cargo-tauri.exe -v
error: 'cargo-tauri' requires a subcommand but one was not provided
  [subcommands: build, dev, icon, info, init, plugin, signer, help]

Usage: cargo-tauri [OPTIONS] <COMMAND>

For more information, try '--help'.
cargo-tauri.exe -V
tauri-cli 1.2.3
cargo.exe
----------
cargo.exe --version
cargo 1.73.0 (9c4383fb5 2023-08-26)
cargo.exe -v
Rust's package manager

Usage: cargo [+toolchain] [OPTIONS] [COMMAND]
       cargo [+toolchain] [OPTIONS] -Zscript <MANIFEST_RS> [ARGS]...

Options:
  -V, --version             Print version info and exit
      --list                List installed commands
      --explain <CODE>      Run `rustc --explain CODE`
  -v, --verbose...          Use verbose output (-vv very verbose/build.rs output)
  -q, --quiet               Do not print cargo log messages
      --color <WHEN>        Coloring: auto, always, never
  -C <DIRECTORY>            Change to DIRECTORY before doing anything (nightly-only)
      --frozen              Require Cargo.lock and cache are up to date
      --locked              Require Cargo.lock is up to date
      --offline             Run without accessing the network
      --config <KEY=VALUE>  Override a configuration value
  -Z <FLAG>                 Unstable (nightly-only) flags to Cargo, see 'cargo -Z help' for details
  -h, --help                Print help

Some common cargo commands are (see all commands with --list):
    build, b    Compile the current package
    check, c    Analyze the current package and report errors, but don't build object files
    clean       Remove the target directory
    doc, d      Build this package's and its dependencies' documentation
    new         Create a new cargo package
    init        Create a new cargo package in an existing directory
    add         Add dependencies to a manifest file
    remove      Remove dependencies from a manifest file
    run, r      Run a binary or example of the local package
    test, t     Run the tests
    bench       Run the benchmarks
    update      Update dependencies listed in Cargo.lock
    search      Search registry for crates
    publish     Package and upload this package to the registry
    install     Install a Rust binary. Default location is $HOME/.cargo/bin
    uninstall   Uninstall a Rust binary

See 'cargo help <command>' for more information on a specific command.
cargo.exe -V
cargo 1.73.0 (9c4383fb5 2023-08-26)

clippy-driver.exe
----------
clippy-driver.exe --version
clippy 0.1.73 (cc66ad46 2023-10-03)
clippy-driver.exe -v
error: no input filename given

clippy-driver.exe -V
clippy 0.1.73 (cc66ad46 2023-10-03)
cross-util.exe
----------
cross-util.exe --version
cross-util 0.2.5 (99b8069 2023-02-12)
cross-util.exe -v
error: unexpected argument '-v' found

  note: to pass '-v' as a value, use '-- -v'

Usage: cross-util.exe [TOOLCHAIN] <COMMAND>

For more information, try '--help'.
cross-util.exe -V
cross-util 0.2.5 (99b8069 2023-02-12)
cross.exe
----------
cross.exe --version
cross 0.2.5 (99b8069 2023-02-12)
[cross] warning: unable to get metadata for package
[cross] note: Falling back to `cargo` on the host.
cargo 1.73.0 (9c4383fb5 2023-08-26)
cross.exe -v
+ cargo metadata --format-version 1
[cross] warning: unable to get metadata for package
        error: could not find `Cargo.toml` in `C:\Users\Max\.cargo\bin` or any parent directory
[cross] note: Falling back to `cargo` on the host.
+ cargo
Rust's package manager

Usage: cargo [+toolchain] [OPTIONS] [COMMAND]
       cargo [+toolchain] [OPTIONS] -Zscript <MANIFEST_RS> [ARGS]...

Options:
  -V, --version             Print version info and exit
      --list                List installed commands
      --explain <CODE>      Run `rustc --explain CODE`
  -v, --verbose...          Use verbose output (-vv very verbose/build.rs output)
  -q, --quiet               Do not print cargo log messages
      --color <WHEN>        Coloring: auto, always, never
  -C <DIRECTORY>            Change to DIRECTORY before doing anything (nightly-only)
      --frozen              Require Cargo.lock and cache are up to date
      --locked              Require Cargo.lock is up to date
      --offline             Run without accessing the network
      --config <KEY=VALUE>  Override a configuration value
  -Z <FLAG>                 Unstable (nightly-only) flags to Cargo, see 'cargo -Z help' for details
  -h, --help                Print help

Some common cargo commands are (see all commands with --list):
    build, b    Compile the current package
    check, c    Analyze the current package and report errors, but don't build object files
    clean       Remove the target directory
    doc, d      Build this package's and its dependencies' documentation
    new         Create a new cargo package
    init        Create a new cargo package in an existing directory
    add         Add dependencies to a manifest file
    remove      Remove dependencies from a manifest file
    run, r      Run a binary or example of the local package
    test, t     Run the tests
    bench       Run the benchmarks
    update      Update dependencies listed in Cargo.lock
    search      Search registry for crates
    publish     Package and upload this package to the registry
    install     Install a Rust binary. Default location is $HOME/.cargo/bin
    uninstall   Uninstall a Rust binary

See 'cargo help <command>' for more information on a specific command.
cross.exe -V
cross 0.2.5 (99b8069 2023-02-12)
[cross] warning: unable to get metadata for package
[cross] note: Falling back to `cargo` on the host.
cargo 1.73.0 (9c4383fb5 2023-08-26)
fd.exe
----------
fd.exe --version
fd 8.6.0
fd.exe -v
error: unexpected argument '-v' found

  note: to pass '-v' as a value, use '-- -v'

Usage: fd.exe [OPTIONS] [pattern] [path]...

For more information, try '--help'.
fd.exe -V
fd 8.6.0
hyperfine.exe
----------
hyperfine.exe --version
hyperfine 1.17.0
hyperfine.exe -v
error: unexpected argument '-v' found

  tip: to pass '-v' as a value, use '-- -v'

Usage: hyperfine.exe [OPTIONS] <command>...

For more information, try '--help'.
hyperfine.exe -V
hyperfine 1.17.0
rg.exe
----------
rg.exe --version
ripgrep 13.0.0
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)
rg.exe -v
error: The following required arguments were not provided:
    <PATTERN>

USAGE:
    
    rg [OPTIONS] PATTERN [PATH ...]
    rg [OPTIONS] -e PATTERN ... [PATH ...]
    rg [OPTIONS] -f PATTERNFILE ... [PATH ...]
    rg [OPTIONS] --files [PATH ...]
    rg [OPTIONS] --type-list
    command | rg [OPTIONS] PATTERN
    rg [OPTIONS] --help
    rg [OPTIONS] --version

For more information try --help

rg.exe -V
ripgrep 13.0.0
rls.exe
----------
rls.exe --version
error: 'rls.exe' is not installed for the toolchain 'stable-x86_64-pc-windows-msvc'
To install, run `rustup component add rls`
rls.exe -v
error: 'rls.exe' is not installed for the toolchain 'stable-x86_64-pc-windows-msvc'
To install, run `rustup component add rls`
rls.exe -V
error: 'rls.exe' is not installed for the toolchain 'stable-x86_64-pc-windows-msvc'
To install, run `rustup component add rls`
rust-analyzer.exe
----------
rust-analyzer.exe --version
rust-analyzer 1.73.0 (cc66ad46 2023-10-03)
rust-analyzer.exe -v
Error: expected initialize request, got error: receiving on an empty and disconnected channel
rust-analyzer.exe -V
unexpected flag: `-V`

rustc.exe
----------
rustc.exe --version
rustc 1.73.0 (cc66ad468 2023-10-03)
rustc.exe -v
error: no input filename given

rustc.exe -V
rustc 1.73.0 (cc66ad468 2023-10-03)
rustdoc.exe
----------
rustdoc.exe --version
rustdoc 1.73.0 (cc66ad468 2023-10-03)
rustdoc.exe -v
error: missing file operand

rustdoc.exe -V
rustdoc 1.73.0 (cc66ad468 2023-10-03)
rustfmt.exe
----------
rustfmt.exe --version
rustfmt 1.6.0-stable (cc66ad46 2023-10-03)
rustfmt.exe -v

rustfmt.exe -V
rustfmt 1.6.0-stable (cc66ad46 2023-10-03)
rustup.exe
----------
rustup.exe --version
rustup 1.26.0 (5af9b9484 2023-04-05)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.73.0 (cc66ad468 2023-10-03)`
rustup.exe -v
rustup 1.26.0 (5af9b9484 2023-04-05)
The Rust toolchain installer

USAGE:
    rustup.exe [OPTIONS] [+toolchain] <SUBCOMMAND>

ARGS:
    <+toolchain>    release channel (e.g. +stable) or custom toolchain to set override

OPTIONS:
    -v, --verbose    Enable verbose output
    -q, --quiet      Disable progress output
    -h, --help       Print help information
    -V, --version    Print version information

SUBCOMMANDS:
    show           Show the active and installed toolchains or profiles
    update         Update Rust toolchains and rustup
    check          Check for updates to Rust toolchains and rustup
    default        Set the default toolchain
    toolchain      Modify or query the installed toolchains
    target         Modify a toolchain's supported targets
    component      Modify a toolchain's installed components
    override       Modify directory toolchain overrides
    run            Run a command with an environment configured for a given toolchain
    which          Display which binary will be run for a given command
    doc            Open the documentation for the current toolchain
    self           Modify the rustup installation
    set            Alter rustup settings
    completions    Generate tab-completion scripts for your shell
    help           Print this message or the help of the given subcommand(s)

DISCUSSION:
    Rustup installs The Rust Programming Language from the official
    release channels, enabling you to easily switch between stable,
    beta, and nightly compilers and keep them updated. It makes
    cross-compiling simpler with binary builds of the standard library
    for common platforms.

    If you are new to Rust consider running `rustup doc --book` to
    learn Rust.
rustup.exe -V
rustup 1.26.0 (5af9b9484 2023-04-05)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.73.0 (cc66ad468 2023-10-03)`
tldr.exe
----------
tldr.exe --version
tealdeer 1.6.1
tldr.exe -v
tealdeer 1.6.1
tldr.exe -V
error: Found argument '-V' which wasn't expected, or isn't valid in this context

	If you tried to supply `-V` as a value rather than a flag, use `-- -V`

USAGE:
    tldr.exe [OPTIONS] [COMMAND]...

For more information try --help
trunk.exe
----------
trunk.exe --version
trunk 0.16.0
trunk.exe -v
trunk 0.16.0
Anthony Dodd <dodd.anthonyjosiah@gmail.com>
Build, bundle & ship your Rust WASM application to the web.

USAGE:
    trunk.exe [OPTIONS] <SUBCOMMAND>

OPTIONS:
        --config <CONFIG>    Path to the Trunk config file [default: Trunk.toml] [env:
                             TRUNK_CONFIG=]
    -h, --help               Print help information
    -v                       Enable verbose logging
    -V, --version            Print version information

SUBCOMMANDS:
    build     Build the Rust WASM app and all of its assets
    clean     Clean output artifacts
    config    Trunk config controls
    help      Print this message or the help of the given subcommand(s)
    serve     Build, watch & serve the Rust WASM app and all of its assets
    watch     Build & watch the Rust WASM app and all of its assets
trunk.exe -V
trunk 0.16.0
xh.exe
----------
xh.exe --version
xh 0.17.0
-native-tls +rustls
xh.exe -v
error: The following required arguments were not provided:
    <[METHOD] URL>

USAGE:
    xh.exe --verbose <[METHOD] URL>

For more information try --help
xh.exe -V
xh 0.17.0
```
</details>

**Comment by 3tilley** [comment]
I've bumped the Cargo.toml and just left it at `--version` for now.

Thanks again for the review!

**Review by dnjstrom — APPROVED** [review]

**Comment by dnjstrom** [comment]
I've cut a new [release now with your changes](https://github.com/dnjstrom/git-select-branch/releases/tag/v0.2.3). Thank you for your contribution! 🙌
